tokio-quiche/src/http3/stats.rs

```diff
@@ -193,4 +209,10 @@ impl H3AuditStats {
     pub fn set_sent_stream_fin(&self, sent_stream_fin: StreamClosureKind) {
         self.sent_stream_fin.store(sent_stream_fin);
     }
+
+    #[inline]
+    pub fn add_header_flush_duration(&self, duration: Duration) {
+        let current = self.headers_flush_duration.load();
+        self.headers_flush_duration.store(current + duration);
```

Comment:
This looks a bit sketchy TBH since the load and store are not atomic, but I think it's actually fine? As in, we are only ever sending one HEADERS at a time for any given stream so this is not going to be called concurrently, but maybe worth adding a comment to avoid future confusion.